doc/builders/special/vm-tools.section.md

```diff
@@ -0,0 +1,116 @@
+# vmTools {#sec-vm-tools}
+
+A set of VM related utilities, that help in building some packages in more advanced scenarios.
+
+## `vmTools.createEmptyImage` {#vm-tools-createEmptyImage}
+
+A bash script fragment that produces a disk image at `destination`.
+
+### Attributes
+
+* `size`. The disk size, in MiB.
+* `fullName`. Name that will be written to `${destination}/nix-support/full-name`.
+* `destination` (optional, default `$out`). Where to write the image files.
+
+## `vmTools.runInLinuxVM` {#vm-tools-runInLinuxVM}
+
+Run a derivation in a Linux virtual machine (using Qemu/KVM). By default, there is no disk image; the root filesystem is
+a `tmpfs`, and the nix store is shared with the host (via the 9P protocol). Thus, any pure Nix derivation should run
+unmodified.
+
+If the build fails and Nix is run with the `-K` option, a script `run-vm` will be left behind in the temporary build
+directory that allows you to boot into the VM and debug it interactively.
+
+### Attributes
+
+* `preVM` (optional). Shell command to be evaluated *before* the VM is started (i.e., on the host).
+* `memSize` (optional, default `512`). The memory size of the VM in megabytes.
+* `diskImage` (optional). A file system image to be attached to `/dev/sda` (Note that currently we expect the image to
+  contain a filesystem, not a full disk image with a partition table etc).
+
+### Examples
+
+```nix
+# Build the derivation hello inside a VM
+runInLinuxVM hello
+
+# Build inside a VM with extra memory
+runInLinuxVM (hello.overrideAttrs (_: { memSize = 1024; }))
+
+# Use VM with a disk image (implicitly sets `diskImage`, see `vmTools.createEmptyImage`)
+runInLinuxVM (hello.overrideAttrs (_: { preVM = createEmptyImage { size = 1024; fullName = "vm-image"; }; }))
+```
```

Comment:
Do not use prose in comments, it's hard to read and to discern from *explanatory* comments as we're used to in actual code. 

Not sure about the scope aspect. While the samples are nice and easy on the eye, they're not self-contained and therefore not usable without contextual knowledge that we cannot assume from beginners (even if you say this is for advanced use cases - no one knows: that may be the first thing someone does with Nix, for whatever reason). 

If we had Sphinx, we could have full working samples and highlight the relevant part. I leave it up to you to decorate every one with `{ pkgs }: with pkgs;` or not.

````suggestion
Build the derivation hello inside a VM:

```nix
runInLinuxVM hello
```

Build inside a VM with extra memory:

```nix
runInLinuxVM (hello.overrideAttrs (_: { memSize = 1024; }))
```

Use VM with a disk image (implicitly sets `diskImage`, see 
[`vmTools.createEmptyImage`](#vm-tools-createEmptyImage)):

```nix
runInLinuxVM (hello.overrideAttrs (_: {
  preVM = createEmptyImage {
    size = 1024;
    fullName = "vm-image";
  };
}))
```
````